cli/utils.go

```diff
@@ -70,6 +73,57 @@ func parseMatchers(inputMatchers []string) ([]labels.Matcher, error) {
 	return matchers, nil
 }
 
+// GetRemoteAlertmanagerConfigStatus returns status responsecontaining configuration from remote Alertmanager
+func GetRemoteAlertmanagerConfigStatus(ctx context.Context, alertmanagerURL *url.URL) (*client.ServerStatus, error) {
+	c, err := api.NewClient(api.Config{Address: alertmanagerURL.String()})
+	if err != nil {
+		return nil, err
+	}
+	statusAPI := client.NewStatusAPI(c)
+	status, err := statusAPI.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return status, nil
+}
+
+func checkRoutingConfigInputFlags(alertmanagerURL *url.URL, configFile string) {
+	if alertmanagerURL != nil && configFile != "" {
+		fmt.Fprintln(os.Stderr, "Warning: --config.file flag overrides the --alertmanager.url.")
+	}
+	if alertmanagerURL == nil && configFile == "" {
+		kingpin.Fatalf("You have to specify one of --config.file or --alertmanager.url flags.")
+	}
+}
+
+func loadAlertmanagerConfig(ctx context.Context, alertmanagerURL *url.URL, configFile string) (*amconfig.Config, error) {
+	checkRoutingConfigInputFlags(alertmanagerURL, configFile)
+	if configFile != "" {
+		cfg, _, err := amconfig.LoadFile(configFile)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+	if alertmanagerURL != nil {
+		status, err := GetRemoteAlertmanagerConfigStatus(ctx, alertmanagerURL)
+		if err != nil {
+			return nil, err
+		}
+		return status.ConfigJSON, nil
+	}
+	return nil, errors.New("Failed to get Alertmanager configuration.")
+}
+
+// ConvertCientToCommonLabelSet converts client.LabelSet to model.Labelset
+func ConvertCientToCommonLabelSet(cls client.LabelSet) model.LabelSet {
```

Comment:
`convertClientToCommonLabelSet`. There's a typo and I don't think this needs to be exported.